drivers/temperature/adt7420/adt7420.c

```diff
@@ -181,14 +154,16 @@ int32_t adt7420_init(struct adt7420_dev **device,
 		dev->resolution_setting = init_param.resolution_setting;
 
 		/* Reset device to default values to ensure all internal circuitry is properly initialised*/
-		adt7420_reset_interface(dev);
+		if (adt7420_reset(dev) != SUCCESS) {
```

Comment:
I see the error code is checked now but there is one more thing to do, pass the return code back - unaltered. There is no reason for us to translate the error code here from -ENOMEM or -EBADMSG or anything like that into FAILURE.

See this as an example, by following the `ret` variable:
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/adc/ad400x/ad400x.c#L173

Reply:
Implemented the return value by a variable holding the actual value returned.